File: csharp/1700.number-of-students-unable-to-eat-lunch.cs
/*
 * @lc app=leetcode id=1700 lang=csharp
 *
 * [1700] Number of Students Unable to Eat Lunch
 */

// @lc code=start
public class Solution {
    public int CountStudents(int[] students, int[] sandwiches) {
        var count = new int[] { 0, 0 };
        int n = students.Length, t = 0;
        foreach (int s in students) {
            ++count[s];
        }
        for (; t < n && count[sandwiches[t]] > 0; ++t) {
            --count[sandwiches[t]];
        }
        return n - t;
    }
}
// @lc code=end

File: java/1700.number-of-students-unable-to-eat-lunch.java
/*
 * @lc app=leetcode id=1700 lang=java
 *
 * [1700] Number of Students Unable to Eat Lunch
 */

// @lc code=start
class Solution {
    public int countStudents(int[] students, int[] sandwiches) {
        int n = students.length;
        int[] count = new int[2];
        for (int s : students) {
            ++count[s];
        }
        int t;
        for (t = 0; t < n && count[sandwiches[t]] > 0; ++t) {
            --count[sandwiches[t]];
        }
        return n - t;
    }
}
// @lc code=end

